### PR TITLE
Feature/crd docs

### PR DESCRIPTION
**Pull request by cloudziu**
This PR is a follow up to [872](https://github.com/berops/claudie/pull/872). 
It is updating the claudie documentation regarding implementing `inputmanifest` custom resource definition.

### CODE REVIEW

File: docs/claudie-workflow/claudie-workflow.md [hunk 1]
```diff
@@ -242,15 +242,15 @@ Kuber manipulates the cluster resources using `kubectl`.
...
 ### API
 
 >This service is a gRPC client, thus it does not provide any API
 
 ### Flow
 
-- User applies a new secret holding a manifest
+- User applies a new inputmanifest crd holding a manifest
 - Frontend detects it and processes the created/modified input manifest
 - Upon deletion of user-created secrets, Frontend initiates a deletion process of the manifest
```
Comment (MiroslavRepka) [inline docs/claudie-workflow/claudie-workflow.md:256]: `user-created secrets` -> `user-created InputManifests`

File: docs/input-manifest/api-reference.md [hunk 2]
```diff
@@ -29,167 +29,139 @@ Manifest is a definition of the user's infrastructure. It contains cloud provide
...
 
-- `gcp` [GCP](#gcp)
-  
-  List of configuration options for [Google Cloud](https://cloud.google.com/). This field is optional.
+- `name`
 
-- `hetzner` [Hetzner](#hetzner)
-  
-  List of configuration options for [Hetzner Cloud](https://www.hetzner.com/cloud) . This field is optional.
+  The name of the provider specyfication. It has to be unique across all providers.
```
Comment (MiroslavRepka) [inline docs/input-manifest/api-reference.md:34]: `specyfication` -> `specification`

Also, I think we call them `provider instances`, so maybe `The name of the provider instance which has to be unique across all providers.`

File: docs/input-manifest/api-reference.md [hunk 3]
```diff
@@ -29,167 +29,139 @@ Manifest is a definition of the user's infrastructure. It contains cloud provide
...
 
-- `oci` [OCI](#oci)
-  
-  List of configuration options for [Oracle Cloud Infrastructure](https://www.oracle.com/uk/cloud/). This field is optional.
+- `providerType`
 
-- `aws` [AWS](#aws)
-  
-  List of configuration options for [Amazon Web Services](https://aws.amazon.com/). This field is optional.
+  Type of a provider. The providerType defines mandatory fields that has to be included for a specyfic provider. A list of available providers can be found at [providers section](./providers). Allowed values are:
```
Comment (MiroslavRepka) [inline docs/input-manifest/api-reference.md:38]: `specific`

File: docs/input-manifest/api-reference.md [hunk 4]
```diff
@@ -29,167 +29,139 @@ Manifest is a definition of the user's infrastructure. It contains cloud provide
...
-  List of HetznerDNS configuration for [HetznerDNS](https://www.hetzner.com/dns-console). This field is optional.
+## SecretRef
+  
+  SecretReference represents a Kubernetes Secret Reference. It has enough information to retrieve secret in any namespace.
 
-Support for more cloud providers is in the [roadmap](https://github.com/berops/claudie/blob/master/docs/roadmap/roadmap.md).
+- `name`
 
-## Cloudflare
+  Name is unique within a namespace to reference a secret resource.
```
Comment (MiroslavRepka) [inline docs/input-manifest/api-reference.md:63]: `Name of the secret, which holds data for the particular cloud provider instance`

File: docs/input-manifest/api-reference.md [hunk 5]
```diff
@@ -29,167 +29,139 @@ Manifest is a definition of the user's infrastructure. It contains cloud provide
...
 
-Collection of data defining Cloudflare provider configuration.
+- `namespace`
 
-To find out how to configure Cloudflare follow the instructions [here](./providers/cloudflare.md)
+  Namespace defines the space within which the secret name must be unique.
```
Comment (MiroslavRepka) [inline docs/input-manifest/api-reference.md:67]: `Namespace of the secret which holds data for the particular cloud provider instance`

File: docs/input-manifest/api-reference.md [hunk 6]
```diff
@@ -29,167 +29,139 @@ Manifest is a definition of the user's infrastructure. It contains cloud provide
...
 
-- `name`
+## Cloudflare
```
Comment (MiroslavRepka) [inline docs/input-manifest/api-reference.md:69]: As these sections are essentially under the `secretRef`, maybe add one more level of headline, i.e. `###`

File: docs/input-manifest/api-reference.md (hunk 3 above)
Comment (katapultcloud) [inline docs/input-manifest/api-reference.md:38]: typo specyfic should be specific

File: docs/input-manifest/providers/hetzner.md [hunk 7]
```diff
@@ -39,89 +46,111 @@ If you wish to use Hetzner as your DNS provider where Claudie creates DNS record
...
+          control:
+            - control-hetzner
+          compute:
+            - compute-1-hetzner
+            - compute-2-hetzner
 ```
 
 ### Multi provider, multi region clusters example
+#### Create a secret for Hetzner provider
+The secret for an Azure provider must include the following mandatory fields: `credentials`.
```
Comment (MiroslavRepka) [inline docs/input-manifest/providers/hetzner.md:130]: `The secret for an Hetzner provider...`

File: docs/input-manifest/api-reference.md (hunk 2 above)
Comment (katapultcloud) [inline docs/input-manifest/api-reference.md:34]: typo specyfication should be specification

File: docs/input-manifest/providers/hetzner.md [hunk 8]
```diff
@@ -39,89 +46,111 @@ If you wish to use Hetzner as your DNS provider where Claudie creates DNS record
 ## Input manifest examples
 
 ### Single provider, multi region cluster example
+#### Create a secret for Hetzner provider
+The secret for an Azure provider must include the following mandatory fields: `credentials`.
```
Comment (MiroslavRepka) [inline docs/input-manifest/providers/hetzner.md:50]: `The secret for an Hetzner provider ...`

File: README.md [hunk 9]
```diff
@@ -106,25 +106,64 @@ For adding support for other cloud providers, open an issue or propose a PR.
...
+    Check the [supported providers](#supported-providers) for input manifest examples. For an input manifest spanning all supported hyperscalers checkout out [this example](https://docs.claudie.io/latest/input-manifest/example.md).
+
+2. Deploy inputmanifest resource which Claudie uses to create infrastructure, include the created secret in `.spec.providers` as follows:
     ```bash
-    kubectl create secret generic input-manifest --from-file=input-manifest.yaml -n claudie -oyaml --dry-run=client | kubectl
-    apply -f -
+    kubectl apply -f - <<EOF
+    apiVersion: claudie.io/v1beta1
+    kind: InputManifest
+    metadata:
```
[the remark this answers is not in the document]
Comment (cloudziu) [reply README.md:125]: InputManifest can be deployed in any namespace - so I would not indicate that it has to be in `claudie` namespace.

[the remark this answers is not in the document]
Comment (MiroslavRepka) [reply README.md:123]: oh, I did not know that, thanks for the clarification!

**Review by MiroslavRepka — APPROVED** [review]
LGTM

**Review by MarioUhrik — APPROVED** [review]
Hi Adam, LGTM

**Pull request merged**